### PR TITLE
HWKINVENT-93 | Metric and Resource type under feed

### PR DESCRIPTION
**Pull request by pavolloffay**

### CODE REVIEW

File: api/src/main/java/org/hawkular/inventory/api/model/Path.java [hunk 1]
```diff
@@ -834,17 +834,23 @@ private boolean fillPossiblePathsToTarget(Class<?> targetType, Class<?> currentT
 
             if (targetType.equals(currentType)) {
                 if (isStart) {
-                    result.add(currentType);
+                    if (!result.contains(currentType)) {
```
Comment (metlos) [inline api/src/main/java/org/hawkular/inventory/api/model/Path.java]: Is this necessary? `result` is a set, so a straight add should actually be cheaper than a check + add - both will try to look up if the element is present.

File: api/src/main/java/org/hawkular/inventory/base/BaseFeeds.java [hunk 2]
```diff
@@ -160,5 +174,17 @@ public Multiple(TraversalContext<BE, Feed> context) {
         public Metrics.ReadContained metrics() {
             return new BaseMetrics.ReadContained<>(context.proceedTo(contains, Metric.class).get());
         }
+
+        @Override
+        public MetricTypes.ReadWrite metricTypes() {
+            // todo
```
Comment (metlos) [inline api/src/main/java/org/hawkular/inventory/base/BaseFeeds.java]: ding, ding :)

File: api/src/main/java/org/hawkular/inventory/api/model/Path.java (hunk 1 above)
Comment (pavolloffay) [inline api/src/main/java/org/hawkular/inventory/api/model/Path.java]: corrected

File: api/src/main/java/org/hawkular/inventory/base/BaseFeeds.java (hunk 2 above)
Comment (pavolloffay) [inline api/src/main/java/org/hawkular/inventory/base/BaseFeeds.java]: Ups... my big mistake.. Implemented to ReadContained.

**Pull request merged**